src/main/java/ip/duke/Duke.java

```diff
@@ -0,0 +1,233 @@
+                if (e.category.equals("event")) {
+                    System.out.println("☹ OOPS!!! The description of an event cannot be empty.");
+                } else {
+                    System.out.println("☹ OOPS!!! The description of a " + e.category + " cannot be empty.");
+                }
+                System.out.println("Please complete the description! :)");
+                printLine();
+            }
+            command = in.nextLine();
+
+        }
+
+        try {
+            PrintWriter writer = new PrintWriter("data/duke.txt");
+            writer.print("");
+            writer.close();
+            for (Task presentTask : list) {
+                appendToFile(presentTask.toDataString());
+                appendToFile(System.lineSeparator());
+            }
+        } catch (FileNotFoundException e){
+            System.out.println("File not found");
+        } catch (IOException e) {
+            System.out.println("Unable to write to file");
+        }
+        printBye();
+    }
+
+    private static void loadData() throws IOException {
+        try {
+            Scanner dataScanner = new Scanner(filePath);
+            while (dataScanner.hasNext()) {
+                String[] dataEntry = dataScanner.nextLine().split("\\|");
+                switch (dataEntry[0]) {
+                case "T":
+                    list.add(new Todo(dataEntry[2]));
+                    break;
+                case "D":
+                    list.add(new Deadline(dataEntry[2], dataEntry[3]));
+                    break;
+                case "E":
+                    list.add(new Event(dataEntry[2], dataEntry[3]));
+                    break;
+                default:
+                    break;
+                }
+                if (dataEntry[1].equals("1")) {
+                    list.get(index).setDone(true);
+                }
+                index++;
+            }
+
+        } catch (FileNotFoundException e) {
+            File dataDirectory = new File("data");
+            dataDirectory.mkdir();
+            File dukeFile = new File("data","duke.txt");
+            dukeFile.createNewFile();
+        }
+    }
+
+    private static void appendToFile(String textToAppend) throws IOException {
+        FileWriter fw = new FileWriter("data/duke.txt", true);
+        fw.write(textToAppend);
+        fw.close();
+    }
+
+
+    public static void printGreeting() {
+        System.out.println("Hello! I'm ip.taskmaster.Duke");
+        System.out.println("What can I do for you?\n");
+        printLine();
+    }
+
+
+    public static void recordTasks(ArrayList<Task> list, int index, String command, String category) {
+        printLine();
+        System.out.println("Got it. I've added this task:");
+        if (category.equals("todo")) {
+            list.add(index, new Todo(command.substring(5)));
+        } else {
+            String Time = command.substring(command.indexOf("/") + 4);
+            if (category.equals("deadline")) {
```

Comment:
you may want to consider storing the special commands such as "deadline", "event", etc to public constants